compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -75,83 +75,73 @@ pub(crate) enum RegionNameHighlight {
 
 impl RegionName {
     pub(crate) fn was_named(&self) -> bool {
+        use RegionNameSource::*;
```

Comment:
I tend to think that these decrease the readability of these functions. We're really not savingany keystrokes, and all it does is conceptually separate useful information. I'd prefer if all of these `use EnumKind::*;` changes are dropped.

Reply:
I think there is a spectrum of these. The more compelling ones are when the enum name is long, and there are lots of repetitions of the enum name on adjacent lines, and removing the enum name lets some multi-line patterns collapse into a single line. The `ConstraintCategory` one is the most compelling in this PR. But I can also see I'm unlikely to convince you, so I'll drop these commits.